src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Stringable;
+
+class Url implements Arrayable, Stringable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public ?string $scheme = null,
+        public ?string $host = null,
+        public ?string $port = null,
+        public ?string $user = null,
+        public ?string $pass = null,
+        public ?string $path = null,
+        public ?string $query = null,
+        public ?string $fragment = null,
+    ) {
+    }
+
+    /**
+     * Parse a URL string into a URL object.
+     *
+     * @param  string  $url
+     * @return static
+     */
+    public static function parse(string $url): static
+    {
+        $components = parse_url($url) ?? [];
+
+        return new static(
+            $components['scheme'] ?? null,
+            $components['host'] ?? null,
+            $components['port'] ?? null,
+            $components['user'] ?? null,
+            $components['pass'] ?? null,
+            $components['path'] ?? null,
+            $components['query'] ?? null,
+            $components['fragment'] ?? null,
+        );
```

Comment:
If `$components` would always be an array, you can just destruct the array thanks to named arguments feature:

```suggestion
        return new static(...$components);
```

Reply:
This is a great suggestion, thanks @tontonsb! I've added this in 🙏 

Let me know what you think of the recent revisions.